tests/random_test.py

```diff
@@ -200,9 +200,11 @@ def testRngRandint(self, dtype):
       self.assertTrue(np.all(samples < hi))
 
   @parameterized.named_parameters(jtu.cases_from_list(
-      {"testcase_name": "_{}".format(dtype), "dtype": np.dtype(dtype).name}
-      for dtype in [np.float32, np.float64]))
+      {"testcase_name": "_{}".format(dtype), "dtype": np.dtype(dtype)}
+      for dtype in [np.float16, np.float32, np.float64]))
   def testNormal(self, dtype):
+    if jtu.device_under_test() == "tpu" and dtype.itemsize < 3:
```

Comment:
`jtu` has a bunch of utilities for determining the types supported by a particular backend. Could we use them instead rather than adding ad hoc logic to each test?

Reply:
No, not as currently written. The issue is not that dtype is unsupported on TPU, the issue is that the implementation of `_random_bits` requires unsigned ints of the same size as `dtype`. So, for example, while TPU supports `float16`, it cannot execute the ``random`` functionality for that dtype.